### DISCUSSION TITLE
Rename/Deprecate IMPERIAL unit system

### DISCUSSION THREAD

**Discussion by epenet** [home-assistant/architecture#820] [Entity Models]
As discussed in https://github.com/home-assistant/architecture/discussions/813, so-called `imperial` units all refer to their US value, which are not always equal to their UK counterpart.

I therefore propose that we remove all references to "imperial" when referring to the unit system in the core to avoid confusion.

**Comment by epenet** [comment]
I propose that we add a specific enum for this:

```python
class BaseUnitSystem(StrEnum):
    """Base unit system."""

    INTERNATIONAL = "international"
    US_CUSTOMARY = "us_customary"
```

Components should then be adjusted to stop using the name (eg. `hass.config.units.name == CONF_UNIT_SYSTEM_IMPERIAL`) and stop using the instance (eg. `hass.config.units == IMPERIAL_SYSTEM`) and use only the new property (eg. `hass.config.units.base_unit_system == BaseUnitSystem.US_CUSTOMARY`)

See preliminary PR: https://github.com/home-assistant/core/pull/80180

After all components have been adjusted to stop using the name or the instance, we should be able to deprecate permanently `CONF_UNIT_SYSTEM_IMPERIAL` and `IMPERIAL_SYSTEM`

**Comment by TomBrien** [comment]
I agree with this on the basis of we should use the correct name for our unit systems or we're going to end in a land of confusion and users will be even more confused

**Comment by frenck** [comment]
I see a lot of small architectural changes for units and systems... but I cannot see the big picture you are trying to achieve. This makes all these one-offs really hard to keep track on and makes it impossible to see where this all is going.

This discussion talks about "international", referring to SI, but we currently have "metric" (which s not the same).  Does that imply you want to drop metric as well?

There is nothing ambiguous about imperial. It is well documented and a worldwide standard. The problem is that our imperial != imperial. Our imperial is US Customary. So, we should look into changing/migrating our base core config to fix that confusion. This proposal doesn't tell anything about how to solve that part.

Furthermore, in the end, it shouldn't matter that much. As a unit system provides a preference/preset for preferred units. Integrations should not be working with unit systems them in the end, they should only care about their units.

**Reply by epenet** [reply]
Even though I wish they didn't, we have a few integrations that check for `hass.config.units.name == CONF_UNIT_SYSTEM_IMPERIAL` or `self.hass.config.units == IMPERIAL_SYSTEM`
We even have a few integrations in core that use a config_flow to offer an option: `[CONF_UNIT_SYSTEM_IMPERIAL, CONF_UNIT_SYSTEM_METRIC]`

I have currently found three ways to solve it for Subaru:
- https://github.com/home-assistant/core/pull/80180
- https://github.com/home-assistant/core/pull/80208
- https://github.com/home-assistant/core/pull/80212

**Reply by epenet** [reply]
I think I found another (better) way to solve ths issue in Subaru: https://github.com/home-assistant/core/pull/80229, which is the last component referencing `IMPERIAL_SYSTEM`
